### PR TITLE
update react 18

### PR DESCRIPTION
**Pull request by joazco**
**What**:
Update to React 18 with Node 16

**Checklist**:

- [x] Documentation updated
- [ ] Tests
- [ ] Ready to be merged
      
I'm not sure for the tests but commit passed
- [x] Added myself to contributors table

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -68,17 +69,17 @@
...
   },
   "peerDependencies": {
-    "@types/react": "^16.9.0 || ^17.0.0",
-    "react": "^16.9.0 || ^17.0.0",
-    "react-dom": "^16.9.0 || ^17.0.0",
-    "react-test-renderer": "^16.9.0 || ^17.0.0"
+    "@types/react": "^16.9.0 || ^17.0.0 || ^18.0.0 || ^18.1.0",
+    "react": "^16.9.0 || ^17.0.0 || ^18.0.0 || ^18.1.0",
+    "react-dom": "^16.9.0 || ^17.0.0 || ^18.0.0 || ^18.1.0",
+    "react-test-renderer": "^16.9.0 || ^17.0.0 || ^18.0.0 || ^18.1.0"
```
Comment (mpeyper) [inline package.json:82]: I think these could just be `^16.9.0 || ^17.0.0 || ^18.0.0` as `^18.1.0` is covered by the `^` on `18.0.0`.

**Review by mpeyper — CHANGES_REQUESTED** [review]
If this works for React 18, I'm not opposed to merging this even though we are [deprecating this library](https://github.com/testing-library/react-hooks-testing-library/issues/849) in favour of the RTL implementation.  The lack of migration path to RTL's `renderHook` has been and ongoing issue and blocker for many people wanting to upgrade to React 18 (for which I am very sorry).

Last I checked, the changes to the `act` behaviour in React 18 made all our async tests fail, so I suspect the same will happen when the CI matrix changes are made, but I'm very happy to be wrong on this.

**Comment by TannerS** [comment]
Update?

**Comment by mpeyper** [comment]
> Update?

https://github.com/testing-library/react-hooks-testing-library?tab=readme-ov-file#a-note-about-react-18-support

**Pull request closed without merging**